Slim/Web/Pages/Search.pm

```diff
@@ -592,7 +591,8 @@ sub fillInSearchResults {
 	# This is very similar to a loop in Slim::Web::Pages::BrowseDB....
 	while (my $obj = $rs->next) {
 
-		$obj->store_column('work' => $obj->work->get_column('title')) if $type eq 'track' && $obj->work->get_column('title');
+		# ensure work name is in the place expected by TitleFormatter so that it uses name and not id
+		$obj->store_column('work' => $obj->work->get_column('title')) if $type eq 'track' && $obj->work && $obj->work->get_column('title');
```

Comment:
As outlined in my previous comment I believe that `->work` is invalid, and that's why you'd have to use `$obj->work->get_column`.

I believe it would be more efficient if you did the check on `$obj->workid` rather than `$obj->work`, as that would work on track record without requiring a lookup in the works table. In particular for the many cases where there's no work object required.

And please make this a regular `if () {}` as the line is getting a bit long.

Reply:
With the `prefetch` back in place, checking `$obj->workid` instead of `$obj->work` is not quite so robust because if tracks.work is not null but the corresponding works row doesn't exist, it fails. I know this shouldn't happen, but it will...

Am I right in assuming that with the `prefetch`, this won't matter so much for efficiency?

Reply:
You could still check `$obj->workid` first. It's my understanding that the object lookup would happen the moment you access `$obj->work`. I'm not 100% certain about that. But in this case checking the `workid` first would avoid that lookup.

I think you could enable logging for `database.sql` in order to see what happens when.

Reply:
I just logged the sql as suggested, and with the prefetch in place, there is only one database access, retrieving all the works data. The check for $obj->work does not go back to the database.

When `$tracksRs` is created. Without the prefetch, sure enough there is a separate SQL statement run in the loop which processes $tracksRs.

I suppose what we've proved is that prefetch works as advertised. 

But there's no harm in adding a prior check of `$obj->workid`, so I'll put it in.

The good news is that I've spent some time testing work-related stuff after adding { join_type => 'left' } to the Tracks work relation, with no ill effects identified (yet!).

I'll push the changes shortly.